### PR TITLE
User migration foundation

### PR DESCRIPTION
**Pull request by come-nc**
Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

### CODE REVIEW

**Comment by Pytal** [comment]
Rebased after https://github.com/nextcloud/user_migration/issues/24 was completed on `main` to resolve failing PHP 7.3 CI

**Comment by Pytal** [comment]
Rebased after https://github.com/nextcloud/user_migration/issues/24 merged to `main` to resolve failing PHP 7.3 CI

**Review by Pytal — APPROVED** [review]
Solid 👌, only the occ stuff is used while the background job as well as the database related code is unused?

**Comment by come-nc** [comment]
> Solid ok_hand, only the occ related stuff is used while the background job as well as the database related code is unused?

Yes maybe we can remove this for now, as the occ commands do not use the background job. We will add the background job back if it’s used is some future development. I will remove them in a commit and we can use git history later to get them back.

File: lib/ExportDestination.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		fwrite($stream, $content);
+		rewind($stream);
+		$this->streamer->addFileFromStream($stream, $path);
+		return true;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function copyFromView(View $view, string $sourcePath, string $destinationPath): bool {
```
Comment (icewind1991) [inline lib/ExportDestination.php:66]: This should use the newer api taking a https://github.com/nextcloud/server/blob/master/lib/public/Files/Folder.php and working from there.

File: lib/IImportSource.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	 */
+	public function getFileAsStream(string $path);
+
+	/**
+	 * Copy files from the export to View
+	 *
+	 * string $sourcePath path in the export archive
+	 * string $destinationPath path to copy to in the view
+	 */
+	public function copyToView(View $view, string $sourcePath, string $destinationPath): bool;
```
Comment (icewind1991) [inline lib/IImportSource.php:54]: same here

**Review by icewind1991 — CHANGES_REQUESTED** [review]

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**